scripts/ci/get-files-touched-by.sh

```diff
@@ -3,7 +3,7 @@ set -e
 
 DIR="$(git rev-parse --show-toplevel)"
 marker=$(mktemp)
-touch "$marker" -r "${BASH_SOURCE[0]}"
+touch "$marker"
```

Comment:
This is the only change that scares me - there were some timing issues here but they might be gone with hosted runners (or something else). Let's just remember this change if things start randomly changing again :)

Reply:
Talked to Jeremy and he assigns all responsibility for this script on you :)

I've verified that `touch {file} -r {referenceFile}` works as designed on ubuntu, even though it does not work on mac (needs to be `touch -r {referenceFile} {file}` instead).

The only timing issue I can imagine is if `find -newer` rounds the compared modification times to a minute (or to a second with a really fast running schema script). But that does not seem to be an issue on my ubuntu docker image.

As written, this was caching all files changed in apps/libs since the clone timestamp. I can't see any other changes we need to include before we run `yarn schemas`. Even NX cache hits should result in newer timestamps.

So yeah, let's see if this breaks anything :P

Reply:
@jeremybarbet traitor!

Reply:
@jeremybarbet traitor!